### DISCUSSION TITLE
Problem building a typescript payment handler, ConfigArgValues, TS2322

### DISCUSSION THREAD

**Discussion by agoransson** [vendurehq/vendure#1311] [Q&A] [answered]
Hello, I'm trying to build a payment handler for use in my vendure installation, however I run into some trouble with typescript - specifically the ConfigArgValues doesn't seem to be parsed properly.

I define the args in a similar way as other payment packages, but I just keep running into the same error.

```
error TS2322: Type 'boolean | Date | ID' is not assignable to type 'boolean'.
  Type 'string' is not assignable to type 'boolean'.
```

I have tried to use other payment packages (stripe, braintree) as a guide when building my own, but I get the same problems regardless of what I do.

These are the args I've defined.

```
    args: {
        isLive: {
            type: 'boolean',
        },
        region: {
            type: 'string',
            label: [{ languageCode: LanguageCode.en, value: 'klarna_region' }],
        },
        username: {
            type: 'string',
            label: [{ languageCode: LanguageCode.en, value: 'klarna_username' }],
        },
        password: {
            type: 'string',
            label: [{ languageCode: LanguageCode.en, value: 'klarna_password' }],
        },
    },
```

Any hints on how to handle this type-problem would be vere appreciated!

Thanks!

**Comment by michaelbromley** [comment]
Hi! What version of TypeScript are you using? And is it just the `isLive` arg that causes the issue?

**Reply by agoransson** [reply]
These are my dependencies from the vendure server package (where I test the payment handler I'm making):

```
  "dependencies": {
    ...
    "@vendure/admin-ui-plugin": "1.4.3",
    "@vendure/asset-server-plugin": "1.4.3",
    "@vendure/core": "1.4.3",
    "@vendure/email-plugin": "1.4.3",
    "mysql": "2.18.1",
    "typescript": "4.1.5"
  },
  "devDependencies": {
    "concurrently": "6.2.1",
    "ts-node": "10.2.1"
  }
```

These are from my payment handler:
```
    "devDependencies": {
        "@vendure/common": "^1.4.3",
        "@vendure/core": "^1.4.3",
        "tslint": "^6.1.3",
        "typescript": "^4.5.4"
    },
    "dependencies": {
    },
```

**Reply by michaelbromley** [reply]
Can you try with TS v4.3.5 and see if the problem remains?

**Comment by agoransson** [comment, marked as answer]
This problems seems fixed when updating.
